src/templates/components/facilityMenu/index.tsx

```diff
@@ -0,0 +1,209 @@
+import clsx from 'clsx'
+import { useState, useEffect, Fragment } from 'react'
+import { MenuItemProps } from '@/types/index'
+import { Menu } from '@/types/dataTypes/drupal/menu'
+
+const MenuItem = ({
+  id,
+  url,
+  title,
+  expanded,
+  enabled,
+  items,
+  depth,
+}: MenuItemProps) => {
+  const [isSelected, setIsSelected] = useState(false)
+  const [rootPath, setRootPath] = useState('')
+
+  // inside a hook so that window is available when the component is mounted
+  useEffect(() => {
+    // sets active state ("open" class)
+    setIsSelected(
+      window.location.pathname === url ||
+        window.location.pathname.slice(0, -1) === url
+    )
+
+    // used to determine which children should show
+    setRootPath(window.location.pathname)
+  }, [])
+
+  // can't do this before the hook logic or it errors
+  if (!enabled) return
+
+  // derive depth booleans
+  const isFirstLevel = depth === 1
+  const isDeeperThanSecondLevel = depth >= 2
+
+  // Calculate the indentation for the child items.
+  const indentation = isDeeperThanSecondLevel ? 20 * (depth - 1) : 20
+
+  // Expanded not selected
+  // const isExpanded = expanded && depth === 2 && !isSelected
+  // Expanded beyond level 2 expanded and selected
+  // const moreThanLevel2SelectedExpanded = expanded && depth > 2 && isSelected
+  const isLevelFourOrDeeper = depth >= 4
+
+  if (isFirstLevel) {
+    return (
+      <li key={id} className={'va-sidenav-level-1'}>
+        <h2
+          className={
+            'va-sidenav-item-label vads-u-font-family--sans va-sidenav-item-label-bold'
+          }
+          style={{
+            paddingLeft: indentation,
+            fontSize: '14px',
```

Comment:
Is this how the other component does this, rather than referencing a class? 

(Am I thinking of old school CSS? :) )

Reply:
i would love to reference a class here, but they do the indentation dynamically based on depth rather than an additional class for each level. https://github.com/department-of-veterans-affairs/vets-website/blob/0d38b0e6edd941d0ae691d8ad77503adf6c251fd/src/platform/site-wide/side-nav/components/NavItemRow.js#L34